packages/playground/src/pages/repo-settings-rules-page.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useState } from 'react'
+import {
+  Button,
+  ButtonGroup,
+  Icon,
+  StackedList,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Text,
+  Textarea,
+  Checkbox
+} from '@harnessio/canary'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+import { FormFieldSet } from '../index'
+import { MessageTheme } from '../components/form-field-set'
+
+const formSchema = z.object({
+  name: z.string().min(1, 'Name is required'),
+  description: z.string().min(1, 'Description is required'),
+  targetPatterns: z.string().min(1, 'Patterns are required'),
+  includePatterns: z.enum(['', '1', '2', '3']),
+  toggleValue: z.boolean(),
+  bypassValue: z.string().optional(),
+  access: z.enum(['1', '2'], {}),
+  defaultBranchValue: z.boolean().optional(),
+  editPermissionsValue: z.boolean().optional(),
+  rules: z.array(
+    z.object({
+      id: z.string(),
+      checked: z.boolean()
+    })
+  )
+})
+export type FormFields = z.infer<typeof formSchema>
+// interface RepoSettingsRulesPageProps {
+//   onFormSubmit?: (data: FormFields) => void
+//   onFormCancel?: () => void
+//   apiError?: string | null
+//   isLoading?: boolean
+//   isSuccess?: boolean
+// }
+const rules = [
+  {
+    id: 'request-approval',
+    label: 'Request approval of new changes',
+    description: 'Require re-approval when there are new changes in the pull request'
+  },
+  {
+    id: 'change-requests',
+    label: 'Require resolution of change requests',
+    description: 'All change requests on a pull request must be resolved before it can be merged'
+  },
+  {
+    id: 'comment-resolution',
+    label: 'Require comment resolution',
+    description: 'All comments on a pull request must be resolved before it can be merged'
+  },
+  {
+    id: 'status-checks',
+    label: 'Require status checks to pass',
+    description: 'Selected status checks must pass before a pull request can be merged'
```

Comment:
![image](https://github.com/user-attachments/assets/1cbccc51-1139-4545-81ed-201b410e47e4)
this one would also be more than a boolean

Reply:
Oh interesting. 

The figma gave options for a boolean. I can include the changes but we would need design approvals for this as well. 

@abhinavrastogi-harness 


<img width="573" alt="Screenshot 2024-10-17 at 4 12 56 PM" src="https://github.com/user-attachments/assets/3bbc0581-a5db-4891-ad62-f6d5cc364873">

Reply:
that wont work with the current api then and it doesnt even make sense if u think about merge strategies as a rule, you would need an input or some selected value cc. @sans-harness  @abhinavrastogi-harness

Reply:
Oh yeah - I am implementing it, its a WIP, I just meant we needed to get design approvals. 

Current Progress~

https://github.com/user-attachments/assets/d1176c32-2eb0-4eb2-ade9-fce1c3c56f8f